tracer/src/Datadog.Trace/Tagging/TagsList.cs

```diff
@@ -35,13 +36,15 @@ public virtual string GetTag(string key)
             return null;
         }
 
-        public virtual void SetTag(string key, string value)
+        public virtual void SetTag(string key, string value) => SetTag(key, new LazyOrString(value));
```

Comment:
This change worries me because of the extra allocations. This method is called when users set arbitrary tags that we have not added a fast path for, and now we are wrapping another object around the `string value` that will need to be garbage collected

Reply:
Agreed. _Two_ new objects are allocated: `LazyOrString` and the `Lazy<string>` inside it.